src/components/facets/FacetReducers.ts

```diff
@@ -46,24 +46,32 @@ const changeFacet = (state: IFacetState, action: (IReduxAction<IReduxActionsPayl
         return state;
     }
 
-    let selected = state.selected;
+    let selected = _.clone(state.selected);
     if (_.some(state.selected, (facetRow: IFacet) => facetRow.name === action.payload.facetRow.name)) {
         const selectedIndex: number = _.findIndex(state.selected, {name: action.payload.facetRow.name});
         if (!selected[selectedIndex].exclude && action.payload.facetRow.exclude) {
-            selected[selectedIndex] = {...selected[selectedIndex], exclude: true};
+            selected[selectedIndex] = {
+                name: selected[selectedIndex].name,
+                formattedName: selected[selectedIndex].formattedName,
+                exclude: true,
+            };
         } else {
             selected = _.reject(state.selected, (facetRow: IFacet, index: number) => index === selectedIndex);
         }
     } else {
         selected = [
-            action.payload.facetRow,

```

Comment:
```
selected = [
 {...action.payload.facetRow},
...state.selected,
];
```

Reply:
It is to not pass the count attr. Maybe I can just do an ..._.omit(action.payload.facetRow, 'count')

Reply:
Then I prefer your solution